runtime/vm/ContinuationHelpers.cpp

```diff
@@ -110,7 +112,11 @@ synchronizeWithConcurrentGCScan(J9VMThread *currentThread, J9VMContinuation *con
 		omrthread_monitor_exit(currentThread->publicFlagsMutex);
 
 		internalAcquireVMAccess(currentThread);
+		continuationObject = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
 	}
+
+	Assert_VM_true((uintptr_t)currentThread == continuation->state);
```

Comment:
this discloses details about how we store carrierId into the state, so it could be another helper, something like

assertContinuationMounted(continuation, currentThread);